ButterflyMX Demo/Controllers/IncomingCallViewController.swift

```diff
@@ -233,5 +208,4 @@ extension IncomingCallViewController: BMXCall.IncomingCallUIInputs {
         CallsService.shared.endCurrentCallKitCall()
```

Comment:
Will "handleCallEnded" be called in case I decline a call from CallKit UI? If yes, then we don't need to call `CallsService.shared.endCurrentCallKitCall` because it in order will call CallKit delegate function ` func provider(_ provider: CXProvider, perform action: CXEndCallAction)` which will call the end call function of the BMXCall.

Reply:
so when `handleCallEnded` is called then you will call `CallsService.shared.endCurrentCallKitCall()` and it will trigger `provider(_ provider: CXProvider, perform action: CXEndCallAction)` again which will call `BMXCallKit.shared.endCall()`, but the call is already ended. This is not good.

There are 2 situations which should be handled correctly in case it's CallKit call:
1. Correct call ending from CallKit UI:
`CallsService.shared.endCurrentCallKitCall()` should not be called in `handleCallEnded ` because it's already needed by the users action.

2. Correct call ending from custom UI:
I think in case it's CallKit call the `CallsService.shared.endCurrentCallKitCall` should be called instead of direct calling of `BMXCallKit.shared.endCall()` in the `hangUpAction` from custom UI, so it will trigger `provider(_ provider: CXProvider, perform action: CXEndCallAction) which will call BMXCallKit.shared.endCall()`.